pkg/eval/testutils.go

```diff
@@ -120,6 +120,31 @@ func (e errWithMessage) matchError(e2 error) bool {
 	return e2 != nil && e.msg == e2.Error()
 }
 
+// An errorMatcher for an ExternalCmdExit error that ignores the Pid member.
+// That is, we only match the command name and raw exit status.
+type errCmdExit struct{ v error }
+
+func (e errCmdExit) Error() string { return fmt.Sprintf("%#v", e.v) }
```

Comment:
This should return a description of what is expected. Since this only matches CmdName and WaitStatus, it should return something like `ExternalCmdExit with CmdName = ... and WaitStatus = ...`

Reply:
SGTM as far as being more explicit about the important attributes of the value. But note that `errCmdExit` is a generic struct that wraps an `error`, not an `ExternalCmdExit`. Which necessitates writing `e.v.(ExternalCmdExit).WaitStatus` rather than `e.WaitStatus`. That is obviously hideous. This is closely related to your comment about the `ThrowsCmdExit` method.

Reply:
I opted to change the `errCmdExit` type to wrap `ExternalCmdExit` rather than `error`. That, and a related change, eliminated most of the ugly reflection code.